test/correctness/introspection.cpp

```diff
@@ -5,20 +5,34 @@ using namespace Halide;
 
 // The check has to go in the Halide namespace, because get_source_location looks for the first thing outside of it
 namespace Halide {
+bool paths_equal(const std::string &path1, const std::string &path2) {
+    int length_delta = path1.size() - path2.size();
```

Comment:
Will `int = size_t - size_t` do the right thing for negative cases in all compiler environments? I confess I forget this part of the C int conversion spec

Reply:
We probably should turn on that set of warnings some year. Fixing now.